terraform/groups/ecs-service/data.tf

```diff
@@ -0,0 +1,47 @@
+data "vault_generic_secret" "stack_secrets" {
+  path = "applications/${var.aws_profile}/${var.environment}/${local.stack_name}-stack"
+}
+
+data "aws_vpc" "vpc" {
+  filter {
+    name   = "tag:Name"
+    values = [local.vpc_name]
+  }
+}
+
+data "aws_ecs_cluster" "ecs-cluster" {
+  cluster_name = "${local.name_prefix}-cluster"
+}
+data "aws_iam_role" "ecs-cluster-iam-role" {
+  name = "${local.name_prefix}-ecs-task-execution-role"
+}
+
+data "aws_lb" "dev-site-lb" {
+  name = "dev-site-${var.environment}-lb"
+}
+data "aws_lb_listener" "dev-site-lb-listener" {
+  load_balancer_arn = data.aws_lb.dev-site-lb.arn
+  port              = 443
+}
```

Comment:
Can we please be consistent with our labelling of the resource blocks? We should be using underscores, rather than hypens, in the labels.

E.g.:
```
data "aws_ecs_cluster" "ecs-cluster" {
  cluster_name = "${local.name_prefix}-cluster"
}
```

Should be:
```
data "aws_ecs_cluster" "ecs_cluster" {
  cluster_name = "${local.name_prefix}-cluster"
}
```

And references updated accordingly.